File: admin_app/lib/api/entity/ChatEntity.dart
import 'package:admin_app/api/entity/CityEntity.dart';
import 'package:admin_app/api/entity/MessageEntity.dart';
import 'package:admin_app/api/entity/UserEntity.dart';
import 'package:admin_app/api/entity/enums/ChatMode.dart';
import 'package:json_annotation/json_annotation.dart';

part 'ChatEntity.g.dart';

@JsonSerializable()
class ChatEntity {
  String id;
  ChatMode mode;
  String name;
  UserEntity? member1;
  UserEntity? member2;
  int unread;
  MessageEntity? lastMessage;
  ChatEntity({required this.id, required this.mode, required this.name, required this.unread, this.lastMessage, this.member1, this.member2});

  factory ChatEntity.fromJson(Map<String, dynamic> json) => _$ChatEntityFromJson(json);
  Map<String, dynamic> toJson() => _$ChatEntityToJson(this);
}

File: server/GNext/src/main/java/com/thedeveloper/gnext/entity/ChatEntity.java
package com.thedeveloper.gnext.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.thedeveloper.gnext.enums.ChatMode;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "chats")
@Data
public class ChatEntity {
    @GeneratedValue(strategy = GenerationType.UUID)
    @Id
    String id;
    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "city", nullable = true,referencedColumnName = "name")
    CityEntity city;
    @Enumerated(EnumType.STRING)
    ChatMode mode;

    String name;
    @ManyToOne(fetch = FetchType.LAZY, optional = true)
    @JoinColumn(name = "member1", nullable = true, referencedColumnName = "phone")
    UserEntity member1;
    @ManyToOne(fetch = FetchType.LAZY, optional = true)
    @JoinColumn(name = "member2", nullable = true, referencedColumnName = "phone")
    UserEntity member2;
    @ManyToOne(fetch = FetchType.EAGER, optional = true)
    CompanyEntity company;
    @Transient
    int unread;
    @JsonManagedReference
    @Transient
    MessageEntity lastMessage;
}
